muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/TranscriptOfRecordsController.java

```diff
@@ -41,7 +45,39 @@
   @Inject
   private SearchProvider searchProvider;
   
+  @Inject
+  private Logger log;
+  
   private static final Pattern UPPER_SECONDARY_SCHOOL_SUBJECT_PATTERN = Pattern.compile("^[A-ZÅÄÖ0-9]+$");
+  
+  @PostConstruct
+  public void init() {
+    String commaSeparatedSubjectsOrder = pluginSettingsController.getPluginSetting("transcriptofrecords", "subjectsOrder");
+    if (commaSeparatedSubjectsOrder == null) {
+      log.log(Level.WARNING, "No subjects order defined for studies vies");
+      return;
+    }
+
+    String[] subjects = commaSeparatedSubjectsOrder.split(",");
+    int i = 0;
+    for (String subject : Arrays.asList(subjects)) {
```

Comment:
I don't think saving a few lines is worth losing the safety benefits of iterator-based for. When indexing an array, there's always a possibility of indexing out of bounds, so I try to minimize indexing.